docs/source/examples/datasets.rst

```diff
@@ -0,0 +1,143 @@
+various instruct templates for common tasks that you can either use or take inspiration
+from for a custom template (see :class:`~torchtune.data._instruct_templates.InstructTemplate`).
+
+For example, let's say you want to format your dataset into the Alpaca instruction
+format. You will need to use :class:`~torchtune.data._instruct_templates.AlpacaInstructTemplate`
+which looks like this:
+
+.. code-block:: python
+
+    "Below is an instruction that describes a task, paired with an input that provides further context. "
+    "Write a response that appropriately completes the request.\n\n"
+    "### Instruction:\n{instruction}\n\n### Input:\n{input}\n\n### Response:\n"
+
+Here is an example of sample that is formatted with :class:`~torchtune.data._instruct_templates.AlpacaInstructTemplate`:
+
+.. code-block:: python
+
+    from torchtune.data import AlpacaInstructTemplate
+
+    sample = {
+        "instruction": "Classify the following into animals, plants, and minerals",
+        "input": "Oak tree, copper ore, elephant",
+    }
+    prompt = AlpacaInstructTemplate.format(sample)
+    print(prompt)
+    # Below is an instruction that describes a task, paired with an input that provides further context.
+    # Write a response that appropriately completes the request.
+    #
+    # ### Instruction:
+    # Classify the following into animals, plants, and minerals
+    #
+    # ### Input:
+    # Oak tree, copper ore, elephant
+    #
+    # ### Response:
+    #
+
+Chat formats
+------------
+Chat formats are similar to instruct templates, except that they format system,
+user, and assistant messages in a list of messages (see :class:`~torchtune.data._chat_formats.ChatFormat`).
+
+Here is an example using the :class:`~torchtune.data._chat_formats.Llama2ChatFormat`:
+
+.. code-block:: python
+
+    from torchtune.data import Llama2ChatFormat, Message
+
+    messages = [
+        Message(
+            role="system",
+            content="You are a helpful, respectful, and honest assistant.",
+        ),
+        Message(
+            role="user",
+            content="I am going to Paris, what should I see?",
+        ),
+        Message(
+            role="assistant",
+            content="Paris, the capital of France, is known for its stunning architecture..."
+        ),
+    ]
+
+    formatted_messages = Llama2ChatFormat.format(messages)
+    print(formatted_messages)
+    # [
+    #     Message(
+    #         role="user",
+    #         content="[INST] <<SYS>>\nYou are a helpful, respectful and honest assistant.\n<</SYS>>\n\n"
+    #         "I am going to Paris, what should I see? [/INST] ",
+    #     ),
+    #     Message(
+    #         role="assistant",
+    #         content="Paris, the capital of France, is known for its stunning architecture..."
+    #     ),
+    # ]
+
+Note that the system message is now incorporated in the user message. If you create custom ChatFormats
+you can also add more advanced behavior.
```

Comment:
We probably need a blurb in the tutorial on how to BYOD and how to read data from a remote file as well?

Reply:
you mean in the first llm finetune tutorial?